NwPluginAPI/Core/Extensions/ItemExtensions.cs

```diff
@@ -0,0 +1,220 @@
+				case ItemType.KeycardZoneManager:
+					return true;
+			}
+			return false;
+		}
+
+		/// <summary>
+		/// Gets a <see cref="ItemBase"/> of a <see cref="ItemType"/>.
+		/// </summary>
+		/// <param name="type"></param>
+		/// <returns>The <see cref="ItemBase"/> or <see langword="null"/> if not found.</returns>
+		public static ItemBase GetItemBase(this ItemType type)
+		{
+			if(InventoryItemLoader.TryGetItem(type, out ItemBase itembase))
+				return itembase;
+
+			return null;
+		}
+
+		/// <summary>
+		/// Gets a random attachments code for the specified <see cref="ItemType"/> representing a set of random attachments for a firearm.
+		/// </summary>
+		/// <param name="firearm">The <see cref="ItemType"/> of the firearm to be searched for random attachments.</param>
+		/// <returns>The attachments code, a <see cref="uint"/> value representing a random set of attachments for the firearm.</returns>
+		public static uint GetRandomAttachments(this ItemType firearm)
+		{
+			if (!firearm.IsFirearm())
+				throw new ArgumentException("The provided argument is not a valid firearm.", nameof(firearm));
+
+			return AttachmentsUtils.GetRandomAttachmentsCode(firearm);
+		}
+
+		/// <summary>
+		/// Applies the specified attachments to the given <see cref="Firearm"/>.
+		/// </summary>
+		/// <param name="firearm">The <see cref="Firearm"/> to which the attachments will be applied.</param>
+		/// <param name="attachmentCode">The attachments code representing the set of attachments to apply.</param>
+		public static void ApllyAttachments(this Firearm firearm, uint attachmentCode)
+		{
+			if (firearm == null)
+				throw new ArgumentNullException(nameof(Firearm), "Firearm is null");
+
+			firearm.ApplyAttachmentsCode(attachmentCode, true);
+
+			FirearmStatusFlags firearmStatusFlags = FirearmStatusFlags.MagazineInserted;
+			if (firearm.HasAdvantageFlag(AttachmentDescriptiveAdvantages.Flashlight))
+			{
+				firearmStatusFlags |= FirearmStatusFlags.FlashlightEnabled;
+			}
+			firearm.Status = new(firearm.AmmoManagerModule.MaxAmmo, firearmStatusFlags, firearm.GetCurrentAttachmentsCode());
+		}
+
+		/// <summary>
+		/// Applies a set of random attachments to the specified <see cref="Firearm"/>.
+		/// </summary>
+		/// <param name="firearm">The <see cref="Firearm"/> to which random attachments will be applied.</param>
+		/// <exception cref="ArgumentNullException">Thrown if the <paramref name="firearm"/> is null.</exception>
+		public static void ApllyRandomAttachments(this Firearm firearm)
+		{
+			if (firearm == null)
+				throw new ArgumentNullException(nameof(Firearm), "Firearm is null");
+
+			firearm.ApplyAttachmentsCode(AttachmentsUtils.GetRandomAttachmentsCode(firearm.ItemTypeId), true);
+
+			FirearmStatusFlags firearmStatusFlags = FirearmStatusFlags.MagazineInserted;
+			if (firearm.HasAdvantageFlag(AttachmentDescriptiveAdvantages.Flashlight))
+			{
+				firearmStatusFlags |= FirearmStatusFlags.FlashlightEnabled;
```

Comment:
Potentially check what the old state of the flashlight is, as this will always enable it if there is a flashlight, or we could pass an argument that decides to enable it, disable it, or leave it as is when a flashlight is present.

Perhaps an enum flag could be introduced to decide these modifications, to avoid having a lot of method parameters, please let me know what you think of that idea.